w2/src/config.py

```diff
@@ -0,0 +1,49 @@
+"""
+mylo: to understand "it",  cut "it" up, then seek patterns in the pieces. E.g. here
+we use cuts for multi- objective, semi- supervised, rule-based explanation.
+(c) Tim Menzies <timm@ieee.org>, BSD-2 license
+
+OPTIONS:
+  -c --cohen  small effect size                 = .35
+  -f --file   csv data file name                = "../data/diabetes.csv"
+  -h --help   show help                         = false
+  -k --k      low class frequency kludge        = 1
+  -m --m      low attribute frequency kludge    = 2
+  -s --seed   random number seed                = 31210
+  -t --todo   start up action                   = help
+"""
+
+import re
+from utils import coerce
+from utils import o
+
+
+
+"""
+In this code, global settings are kept in `the` (which is parsed from `__doc__`).
+This variable is a `slots`, which is a neat way to represent dictionaries that
+allows easy slot access (e.g. `d.bins` instead of `d["bins"]`)
+"""
+
+class SLOTS(dict): 
+    __getattr__ = dict.get; __setattr__ = dict.__setitem__; __repr__ = o
+
+def get_default_config():
+    the = SLOTS(**{m[1]: coerce(m[2]) for m in re.finditer(r"--(\w+)[^=]*=\s*(\S+)", __doc__)})
+    return the
+
+
+if __name__ == '__main__':
+    # Only for testing
+    the = get_default_config()
+    the.k = 22
+    the.k += 1
```

Comment:
what is the point of this again. I'm confused

Reply:
It's only for testing and doesn't have meaningful perpose, as it will only be executed when we run the script directly (Such as "python3 config.py").